source/extensions/filters/network/client_ssl_auth/BUILD

```diff
@@ -16,6 +16,7 @@ envoy_cc_library(
     name = "client_ssl_auth",
     srcs = ["client_ssl_auth.cc"],
     hdrs = ["client_ssl_auth.h"],
+    category = "envoy.filters.network",
```

Comment:
...this to mark categories in extension build files

Reply:
Right. This will allow you to add the extension category to the metadata in the extension DB built by https://github.com/envoyproxy/envoy/blob/main/docs/generate_extension_db.py.

https://github.com/envoyproxy/envoy/blob/main/tools/protodoc/protodoc.py can be extended to indicate category on the extension metadata formatting.

If you want to build a list of extensions for a given category by Bazel path, I think that's easy enough to do with `bazel query` and the compulsory category field. It's a bit trickier to map from category to configuration proto or configuration RST. One idea might be to extend the type database, built with https://github.com/envoyproxy/envoy/blob/main/tools/type_whisperer/typedb_gen.py to also indicate the extension that a given proto type is responsible for configuring, but the downside is that this takes a very general type database and starts to special case it for extensions (I'm OK with this though).

There are other hackier possibilities. E.g. building all the docs and scraping with grep etc. the extension name from RST and figuring out which path this RST is at, building some RST mapping database.

Reply:
atm i have used the generated extension db to generate an extension type db that can then be used in protodoc

this is a bit docs specific, so i might need to rethink to accomodate generating (vscode) jsonschemas but for now this should give us the abillity to back/forward ref extensions <> types